File: 0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        dp = [cost[0], cost[1]]

        for i in range(2, len(cost)):
            temp = min(dp[0], dp[1]) + cost[i]
            dp[0], dp[1] = dp[1], temp

        return min(dp)

File: 0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.cpp
class Solution {
public:
    const int minCostClimbingStairs(const vector<int>& cost) {
        int a = cost[0], b = cost[1];
        for (int i = 2; i < cost.size(); ++i) {
            int temp = cost[i] + min(a, b);
            a = b;
            b = temp;
        }
        return min(a,b);
    }
};